hcledit.go

```diff
@@ -99,11 +100,17 @@ func (h *HCLEditor) Read(queryStr string, opts ...Option) (map[string]interface{
 		Mode:    walker.Read,
 	}
 
-	if err := w.Walk(h.writeFile.Body(), queries, 0, []string{}); err != nil {
+	err = w.Walk(h.writeFile.Body(), queries, 0, []string{})
+	if err != nil && !fallback {
 		return nil, err
 	}
 
-	return convert(results)
+	ret, convertErr := convert(results)
+	if convertErr != nil {
+		return ret, convertErr
+	}
+
+	return ret, err
```

Comment:
IIUC we want to return the error of `w.Walk` here when fallback occurred. How about naming the error to indicate it comes from `Walk` for readability?

```go
	walkErr := w.Walk(h.writeFile.Body(), queries, 0, []string{})
	if walkErr != nil && !fallback {
		return nil, walkErr
	}

	ret, err := convert(results)
	if err != nil {
		return ret, err
	}

	// When fallback is true, return an error from `w.Walk` to signal that
	// fallback occurred.
	return ret, walkErr
```